poutyne/framework/callbacks/wandb_logger.py

```diff
@@ -0,0 +1,256 @@
+        log_gradient_frequency(int): log gradients and parameters every N batches (Default value = None).
+        training_batch_shape(tuples): Shape of a training batch. Used for logging architecture on wandb
+    Example:
+        .. code-block:: python
+
+            wandb_logger = WandBLogger(
+                                        name="First_run",
+                                        project="Test_project",
+                                        save_dir="/absolute/path/to/directory",
+                                        experiment="First experiment"
+                                       )
+            wandb_logger.log_config_params(config_params=cfg_dict) # logging the config dictionary
+            # our Poutyne experiment
+            experiment = Experiment(
+                                     directory=saving_directory,
+                                     network=network,
+                                     device=device,
+                                     optimizer=optimizer,
+                                     loss_function=cross_entropy_loss,
+                                     batch_metrics=[accuracy]
+                                    )
+            # Using the WandB logger callback during training
+            experiment.train(train_generator=train_loader, valid_generator=valid_loader, epochs=1,
+                             seed=42, callbacks=[wandb_logger])
+            # You can access the wandb run via the attribute .run if you want to use other wandb features
+            image = wandb.Image(an_image, caption="a caption")
+            wandb_logger.run.log({"an example": image})
+    """
+
+    def __init__(
+        self,
+        *,
+        name: Optional[str] = None,
+        group: Optional[str] = None,
+        config: Optional[Dict] = None,
+        save_dir: Optional[str] = None,
+        offline: Optional[bool] = False,
+        run_id: Optional[str] = None,
+        anonymous: Optional[bool] = None,
+        version: Optional[str] = None,
+        project: Optional[str] = None,
+        experiment=None,
+        batch_granularity: Optional[bool] = False,
+        checkpoints_path: Optional[str] = None,
+        initial_artifacts_paths: Optional[List[str]] = None,
+        log_gradient_frequency: Optional[int] = None,
+        training_batch_shape: Optional[tuple] = None,
+    ) -> None:
+        # pylint: disable-msg=too-many-locals
+
+        super().__init__(batch_granularity=batch_granularity)
+
+        if wandb is None:
+            raise ImportError("WandB needs to be installed to use this callback.")
+
+        anonymous_lut = {True: "allow", False: None}
+        self._wandb_init = dict(
+            name=name,
+            group=group,
+            config=config,
+            project=project,
+            id=version or run_id,
+            dir=save_dir,
+            resume="allow",
+            anonymous=anonymous_lut.get(anonymous, anonymous),
```

Comment:
```suggestion
            anonymous=anonymous_lut.get(anonymous),
```
If the goal was to put None in anonymous, you don't need to put a default.